### PR TITLE
[script] [craft] Updated enchanting section to check for item before purchasing.

### PR DESCRIPTION
**Pull request by Dartellum**
This should resolve Issue https://github.com/rpherbig/dr-scripts/issues/5071

### CODE REVIEW

File: craft.lic [hunk 1]
```diff
@@ -298,8 +298,8 @@ class Craft
     if rank <= 25 # Tier 1  Extremely Easy
       # Buy an abolution sigil and wood totem
       ensure_copper_on_hand(2000, @settings)
-      DRCT.order_item(crafting_data['artificing'][@hometown]['stock-room'], 2)
-      DRCT.order_item(crafting_data['artificing'][@hometown]['stock-room'], 6)
+      DRCT.order_item(crafting_data['artificing'][@hometown]['stock-room'], 2) unless DRCI.exists?("abolition sigil-scroll")
+      DRCT.order_item(crafting_data['artificing'][@hometown]['stock-room'], 6) unless DRCI.exists?("bone totem")
```
Comment (rpherbig) [inline craft.lic:302]: The comment on line 299 says 'wood totem', but the string here is 'bone totem'. Should these match?

Same with the other totem changes below.

File: craft.lic [hunk 2]
```diff
@@ -342,11 +342,11 @@ class Craft
     elsif rank <= 300 # Tier 5 - Basic
       # Buy an abolution and induction sigils and wood totem
       ensure_copper_on_hand(2000, @settings)
-      DRCT.order_item(crafting_data['artificing'][@hometown]['stock-room'], 1)
-      DRCT.order_item(crafting_data['artificing'][@hometown]['stock-room'], 2)
+      DRCT.order_item(crafting_data['artificing'][@hometown]['stock-room'], 1) unless DRCI.exists?("induction sigil-scroll")
+      DRCT.order_item(crafting_data['artificing'][@hometown]['stock-room'], 2) unless DRCI.exists?("abolition sigil-scroll")
```
Comment (rpherbig) [inline craft.lic:346]: Should this be 'abolution' per line 343, or is the typo up there?

File: craft.lic (hunk 1 above)
Comment (Dartellum) [reply craft.lic:302]: I noticed that and not sure why it says wood totem:
```
2021-08-08 11:51:29 -0400:>order
2021-08-08 11:51:29 -0400:You see the following items for sale in the catalog:
2021-08-08 11:51:29 -0400:1).  an intricate induction sigil-scroll...... 625 Kronars
2021-08-08 11:51:29 -0400:2).  an intricate abolition sigil-scroll...... 625 Kronars
2021-08-08 11:51:29 -0400:3).  an intricate congruence sigil-scroll..... 312 Kronars
2021-08-08 11:51:29 -0400:4).  an intricate permutation sigil-scroll.... 625 Kronars
2021-08-08 11:51:29 -0400:5).  an intricate rarefaction sigil-scroll.... 1,500 Kronars
2021-08-08 11:51:29 -0400:6).  a bone totem............................. 125 Kronars
```

File: craft.lic (hunk 2 above)
Comment (Dartellum) [reply craft.lic:346]: Let me check that.

Comment (Dartellum) [reply craft.lic:346]: Abolition is correct.

**Comment by Dartellum** [comment]
I am testing adding additional variables to dispose_trash call. It can handle worn_trashcan so testing to pass in that variable.

**Comment by Dartellum** [comment]
Without worn_trashcan set:
```
2021-08-08 15:46:29 -0400:--- Lich: enchant has exited.
2021-08-08 15:46:30 -0400:[craft]>drop my runestone
2021-08-08 15:46:30 -0400:You drop a basic runestone.
```
With:
```
2021-08-08 15:40:34 -0400:>;e DRCI.dispose_trash('runestone', @settings.worn_trashcan, @settings.worn_trashcan_verb)
2021-08-08 15:40:34 -0400:--- Lich: exec1 active.
2021-08-08 15:40:34 -0400:[exec1]>put my runestone in my bucket
2021-08-08 15:40:34 -0400:You drop a colorful elbaite runestone into a sturdy bucket.  Glancing inside, you notice the item being swirled in a whirlpool before dropping to the bottom of the bucket.
2021-08-08 15:40:34 -0400:[exec1]>pull my bucket
2021-08-08 15:40:35 -0400:You pull a lever on the side of your bucket.  It seems to be jammed.
2021-08-08 15:40:35 -0400:[OOC: PULL the bucket again within the next 30 seconds to flush it.].
2021-08-08 15:40:35 -0400:[exec1]>pull my bucket
2021-08-08 15:40:35 -0400:You pull a lever on the side of your bucket.  It seems to be jammed.
2021-08-08 15:40:35 -0400:After a moment, a dull *THUD* echoes from within the bucket.
2021-08-08 15:40:35 -0400:--- Lich: exec1 has exited.
```

**Comment by rpherbig** [comment]
Changes look good to me ✔️ 

Let me know when you're ready for this to be merged

**Comment by Dartellum** [comment]
I ran through a bunch of times and works as expected. I feel it is ready.

**Pull request merged**